src/account-mgr.cpp

```diff
@@ -207,3 +204,41 @@ Account AccountManager::getAccountBySignature(const QString& account_sig) const
 
     return Account();
 }
+
+void AccountManager::updateServerInfo()
+{
+    for (size_t i = 0; i < accounts_.size(); i++)
+        updateServerInfo(i);
+}
+
+void AccountManager::updateServerInfo(unsigned index)
+{
+    ServerInfoRequest *request;
+    // request is taken owner by Account object
+    request = accounts_[index].createServerInfoRequest();
+    connect(request, SIGNAL(success(const Account&, const ServerInfo &)),
+            this, SLOT(serverInfoSuccess(const Account&, const ServerInfo &)));
+    connect(request, SIGNAL(failed(const ApiError&)),
+            this, SLOT(serverInfoFailed(const ApiError&)));
+    request->send();
+}
+
+
+void AccountManager::serverInfoSuccess(const Account &account, const ServerInfo &info)
+{
+    for (size_t i = 0; i < accounts_.size(); i++) {
+        if (accounts_[i] == account) {
+            if (i == 0)
+                emit beforeAccountChanged();
+            accounts_[i].serverInfo = info;
+            if (i == 0)
+                emit accountsChanged();
```

Comment:
Certainly, triggering `accountsChanged` signal again will require a lot of updates on all components. But if we use extra signals, it will involve a lot of complexities when you need to detect whether it has pro feature, and add extra code there. I cannot see it is great for programming and code simplicity. In short, although triggering `accountsChanged` is costly, it is worthy and I prefer to using it.